pkg/genericcomp/genericcomp_test.go

```diff
@@ -0,0 +1,226 @@
+}
+
+// TestNewService tests that a service is created
+// GIVEN a generic component definition
+//  WHEN I call NewService
+//  THEN the expected generic component service should be created
+func TestNewService(t *testing.T) {
+	assert := assert.New(t)
+
+	labels := map[string]string{
+		constants.K8SAppLabel:       constants.VerrazzanoGroup,
+		constants.VerrazzanoBinding: "test-binding",
+		constants.VerrazzanoCluster: "cluster1",
+	}
+
+	selector := map[string]string{
+		GenericComponentSelectorLabel: generic.Name,
+	}
+
+	service := NewService(generic, "test-namespace", labels)
+	assert.Equal("test-generic", service.Name, "service name not equal to expected value")
+	assert.Equal("test-namespace", service.Namespace, "service namespace not equal to expected value")
+	assert.Equal(labels, service.Labels, "service labels not equal to expected value")
+	assert.Equal(selector, service.Spec.Selector, "service selector not equal to expected value")
+	assert.Equal(corev1.ServiceTypeClusterIP, service.Spec.Type, "service type not equal to expected value")
+	assert.Equal(1, len(service.Spec.Ports), "service port list length not equal to expected value")
+	assert.Equal("test-port", service.Spec.Ports[0].Name, "service port name not equal to expected value")
+	assert.Equal(int32(8095), service.Spec.Ports[0].Port, "service port name not equal to expected value")
+	assert.Equal(corev1.ProtocolTCP, service.Spec.Ports[0].Protocol, "service port name not equal to expected value")
+
+	// No container ports specified then no new service is needed.
+	generic.Deployment.Containers[0].Ports = []corev1.ContainerPort{}
+	service = NewService(generic, "test-namespace", labels)
+	assert.Nil(service, "service should be nil")
+}
+
+// TestGetSecrets tests that a list of secrets is returned
+// GIVEN a deployment specification
+//  WHEN I call GetSecrets
+//  THEN the secrets referenced by the deployment are returned
+func TestGetSecrets(t *testing.T) {
+	assert := assert.New(t)
+
+	labels := map[string]string{
+		constants.K8SAppLabel:       constants.VerrazzanoGroup,
+		constants.VerrazzanoBinding: "test-binding",
+		constants.VerrazzanoCluster: "cluster1",
+	}
+
+	deploy := NewDeployment(generic, "test-namespace", labels)
+
+	secrets := GetSecrets(*deploy)
+	assert.Equal(3, len(secrets), "secrets list length not equal to expected value")
+	assert.Equal("test-secret1", secrets[0], "secret not equal to expected value")
+	assert.Equal("test-secret-ref1", secrets[1], "secret not equal to expected value")
+	assert.Equal("test-secret-ref2", secrets[2], "secret not equal to expected value")
+}
+
+// TestUpdateEnvVars tests env variables are added to a generic component deployment
+// GIVEN a managed cluster struct, a generic component name, and a list of environment variables to add
+//  WHEN I call UpdateEnvVars
+//  THEN the environment variables are added to a generic component deployment
+func TestUpdateEnvVars(t *testing.T) {
+	assert := assert.New(t)
```

Comment:
Can you add a test with an existing env var that doesn't get modified, and add a single new one, then check that both exist.

Reply:
Done.